### PR TITLE
Init commit of linux system tests

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## What does this PR do?

This adds system tests to (most) of the linux data streams, which should help in the future with catching low-hanging bugs.

However, we may want to look into expanding the test architecture somewhat, and I'd like input from @mtojek before this gets merged.  The system integration is a bit odd, in that 90% of the code can be covered by pointed at a test file, without any external services, containers, etc. If we want to expand the system tests used in `linux` and `system` further, we need some kind of system that populates a test directory inside the `agent` container with a handful of files that the data streams can read from. I struggled to hack up a solution with what we have now, but decided it wasn't quite worth the ugliness. The system also seems to require test container service...so I had to make a dummy one.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.

## How to test this PR locally

- Pull down PR
- run `elastic-package -v test system` in the `linux` directory

### CODE REVIEW

**Comment by exekias** [comment]
> we need some kind of system that populates a test directory inside the `agent` container with a handful of files that the data streams can read from

Could you elaborate on this? Are you talking about `/proc`&`/sys` folders, or is this something else?

**Comment by jsoriano** [comment]
> we need some kind of system that populates a test directory inside the `agent` container with a handful of files that the data streams can read from

For system tests of metrics so far I think that we are only checking that metrics are collected without errors, we are not checking their values. For this case it may be also enough with checking that it collects any data, so the actual `/proc` and `/sys` filesystems available in the agent container would serve.

For logs you can use `${SERVICE_LOGS_DIR}`.

File: packages/linux/data_stream/conntrack/fields/ecs.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+- name: ecs.version
```
Comment (mtojek) [inline packages/linux/data_stream/conntrack/fields/ecs.yml:1]: Please use dependency management feature to import ECS definitions: [docs](https://github.com/elastic/elastic-package/blob/master/docs/howto/dependency_management.md)

**Comment by fearful-symmetry** [comment]
> Could you elaborate on this? Are you talking about /proc&/sys folders, or is this something else?
@exekias 

Yes, along with a few other things. Depending on the OS config, some procfs files (such as RAID) won't contain any useful information for the metricset to parse. Some, such as cgroup files, won't exist at all inside a container. I spent some time hacking at the container orchestration that exists now to do that, but it was awfully ugly and I decided not to. At minimum, it would be nice to omit the "auxillary" container that `elastic-package test system` seems to require, as it's not actually needed here.

**Comment by exekias** [comment]
> For system tests of metrics so far I think that we are only checking that metrics are collected without errors, we are not checking their values. For this case it may be also enough with checking that it collects any data, so the actual `/proc` and `/sys` filesystems available in the agent container would serve.

I'm very much onboard with this for the initial batch of tests (this PR). It won't be testing the full package but many parts of it are still valid.

On top of that (maybe later), we could also have a `testdata` folder mounted into the agent, the same way we have one for logs, right? This way you could mock a raid file structure and configure the agent to retrieve it, for example.

@mtojek @jsoriano any thoughts on the need for a dummy container when there is no need for it? I'm not particularly concerned with its need by I see @fearful-symmetry point

**Comment by mtojek** [comment]
This is a specific use case where you don't need to deploy anything. Definitely not a blocker, but could be considered as a feature. Please open an issue in elastic-package for this.

**Review by mtojek — COMMENTED** [review]
As you introduce the dependency on ECS, could you please review and adjust also other data streams? I see there are potential candidates, not sure about others:

https://github.com/fearful-symmetry/integrations/blob/linux-system-tests/packages/linux/data_stream/users/fields/ecs.yml
https://github.com/fearful-symmetry/integrations/blob/linux-system-tests/packages/linux/data_stream/socket/fields/ecs.yml

**Comment by fearful-symmetry** [comment]
Alright, added the ECS fields to the other data streasm. @mtojek / @exekias I agree this isn't a blocker, just wanted to make the shortcomings known here. I'll open an issue on `elastic-package`.

File: packages/linux/data_stream/conntrack/fields/ecs.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  external: ecs
+- name: error.message
```
Comment (mtojek) [inline packages/linux/data_stream/conntrack/fields/ecs.yml:3]: Usually we don't add `error.message` fields. Is there any reason why you added it here? I mean, are there any problems with system tests? If so, then they should be fixed first.

**Review by mtojek — APPROVED** [review]
LGTM, but I left one comment regarding `error.message` to double-check.

File: packages/linux/data_stream/conntrack/fields/ecs.yml (hunk 2 above)
Comment (fearful-symmetry) [reply packages/linux/data_stream/conntrack/fields/ecs.yml:3]: Ah, that was probably just me copying the ECS fields from other data streams. I can remove it.

**Pull request merged**